packages/api/cms-api/src/page-tree/createPageTreeResolver.ts

```diff
@@ -151,12 +128,47 @@ export function createPageTreeResolver({
         }
 
         @ResolveField(() => pageContentUnion, { nullable: true })
-        async document(@Parent() node: PageTreeNodeInterface): Promise<typeof pageContentUnion | undefined> {
-            const activeDocument = await this.pageTreeService.getActiveAttachedDocument(node.id, node.documentType);
-            if (!activeDocument) {
-                return null;
+        async document(@Parent() node: PageTreeNodeInterface, @Info() info: GraphQLResolveInfo): Promise<typeof pageContentUnion | undefined> {
+            if (info.fieldNodes.length === 1) {
```

Comment:
Can't we use dataloader for documents too?

I don't fully understand what you're doing here. Does this code check if `document` has already been loaded for the query and doesn't load it again? Does this ever happen, that document is loaded twice?

Reply:
Okay, this is really an advanced optimization. Should we add a comment explaining the intention for future reference? Or is the code enough?